From 24b8d8262526da9391f04a696264ca9e63bb7d9d Mon Sep 17 00:00:00 2001
From: Saleem Abdulrasool <compnerd@compnerd.org>
Date: Sat, 7 Aug 2010 16:28:28 -0700
Subject: [PATCH 1/2] allow building of only gudev

---
 Makefile.am  |   23 ++++++++++++++---------
 configure.ac |   20 ++++++++++++++++----
 2 files changed, 30 insertions(+), 13 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 3406c3a..8d5842e 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -292,15 +292,9 @@ extras_v4l_id_v4l_id_LDADD = libudev/libudev-private.la
 libexec_PROGRAMS += extras/v4l_id/v4l_id
 dist_udevrules_DATA += extras/v4l_id/60-persistent-v4l.rules
 
-if ENABLE_EXTRAS
-# ------------------------------------------------------------------------------
-# conditional extras (need glib, libusb, libacl, ...)
-# ------------------------------------------------------------------------------
-dist_udevrules_DATA += \
-	rules/rules.d/75-net-description.rules \
-	rules/rules.d/75-tty-description.rules \
-	rules/rules.d/78-sound-card.rules
+BUILT_SOURCES =
 
+if ENABLE_GUDEV
 # ------------------------------------------------------------------------------
 # GUdev - libudev gobject interface
 # ------------------------------------------------------------------------------
@@ -338,7 +332,7 @@ dist_extras_gudev_libgudev_1_0_la_SOURCES = \
 	extras/gudev/gudevmarshal.c \
 	extras/gudev/gudevenumtypes.h \
 	extras/gudev/gudevenumtypes.c
-BUILT_SOURCES = $(dist_extras_gudev_libgudev_1_0_la_SOURCES)
+BUILT_SOURCES += $(dist_extras_gudev_libgudev_1_0_la_SOURCES)
 
 extras_gudev_libgudev_1_0_la_CPPFLAGS = \
 	$(AM_CPPFLAGS) \
@@ -416,6 +410,17 @@ typelibs_DATA = extras/gudev/GUdev-1.0.typelib
 CLEANFILES += $(gir_DATA) $(typelibs_DATA)
 endif # ENABLE_INTROSPECTION
 
+endif # ENABLE_GUDEV
+
+if ENABLE_EXTRAS
+# ------------------------------------------------------------------------------
+# conditional extras (need glib, libusb, libacl, ...)
+# ------------------------------------------------------------------------------
+dist_udevrules_DATA += \
+	rules/rules.d/75-net-description.rules \
+	rules/rules.d/75-tty-description.rules \
+	rules/rules.d/78-sound-card.rules
+
 # ------------------------------------------------------------------------------
 # Bluetooth HID devices with special magic to switch the device
 # ------------------------------------------------------------------------------
diff --git a/configure.ac b/configure.ac
index b9b1b0b..a52526e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -46,6 +46,18 @@ if test "x$enable_logging" = "xyes"; then
 	AC_DEFINE(ENABLE_LOGGING, [1], [System logging.])
 fi
 
+AC_ARG_ENABLE(gudev,
+              AS_HELP_STRING([--disable-gudev], [disable libgudev - libudev gobject interface]),,
+              enable_gudev=yes)
+
+if test x"$enable_gudev" = x"yes" ; then
+   PKG_CHECK_MODULES([GLIB], [glib-2.0 >= 2.7.0 gobject-2.0 >= 2.7.0])
+   AC_SUBST([GLIB_CFLAGS])
+   AC_SUBST([GLIB_LIBS])
+fi
+
+AM_CONDITIONAL([ENABLE_GUDEV], [test x"$enable_gudev" = x"yes"])
+
 AC_ARG_WITH(firmware-path,
 	AS_HELP_STRING([--with-firmware-path=DIR[[[:DIR[...]]]]],
 	   [Firmware search path (default=/lib/firmware/updates:/lib/firmware)]),
@@ -78,10 +90,6 @@ if test "x$enable_extras" = xyes; then
 		AC_MSG_ERROR([gperf is needed])
 	fi
 
-	PKG_CHECK_MODULES([GLIB], [glib-2.0 >= 2.7.0 gobject-2.0 >= 2.7.0])
-	AC_SUBST([GLIB_CFLAGS])
-	AC_SUBST([GLIB_LIBS])
-
 	AC_CHECK_LIB([acl], [acl_init], [:], AC_MSG_ERROR([libacl not found]))
 	AC_CHECK_HEADER([acl/libacl.h], [:], AC_MSG_ERROR([libacl header not found]))
 
@@ -118,6 +126,10 @@ AC_ARG_ENABLE([introspection],
 	AS_HELP_STRING([--disable-introspection], [disable GObject introspection]),
 	[], [enable_introspection=yes])
 if test "x$enable_introspection" = xyes; then
+	if test x"$enable_gudev" != x"yes" ; then
+		AC_MSG_ERROR(["introspection requires that gudev is built"])
+	fi
+
 	PKG_CHECK_MODULES([INTROSPECTION], [gobject-introspection-1.0 >= 0.6.2])
 	AC_DEFINE([ENABLE_INTROSPECTION], [1], [enable GObject introspection support])
 	AC_SUBST([G_IR_SCANNER], [$($PKG_CONFIG --variable=g_ir_scanner gobject-introspection-1.0)])
-- 
1.7.1

