diff -urp libspiro/spiro.c libspiro-inkscape/spiro.c
--- libspiro/spiro.c	2007-10-29 00:18:32.000000000 +0000
+++ libspiro-inkscape/spiro.c	2010-08-01 07:40:06.257710200 +0000
@@ -731,6 +731,10 @@ spiro_iter(spiro_seg *s, bandmat *m, int
 	add_mat_line(m, v, derivs[1][1], -ends[1][1], 1, j, jk0r, jinc, nmat);
 	add_mat_line(m, v, derivs[2][1], -ends[1][2], 1, j, jk1r, jinc, nmat);
 	add_mat_line(m, v, derivs[3][1], -ends[1][3], 1, j, jk2r, jinc, nmat);
+	if (jthl >= 0)
+	    v[jthl] = mod_2pi(v[jthl]);
+	if (jthr >= 0)
+	    v[jthr] = mod_2pi(v[jthr]);
 	j += jinc;
     }
     if (cyclic) {
@@ -774,6 +778,7 @@ spiro_iter(spiro_seg *s, bandmat *m, int
 	    s[i].ks[k] += dk;
 	    norm += dk * dk;
 	}
+	s[i].ks[0] = 2 * mod_2pi(s[i].ks[0] / 2.);
     }
     return norm;
 }
