diff -Naur libcap-2.16-orig/Make.Rules libcap-2.16/Make.Rules
--- libcap-2.16-orig/Make.Rules	2008-12-04 07:03:30.000000000 +0000
+++ libcap-2.16/Make.Rules	2008-12-24 22:16:31.000000000 +0000
@@ -43,12 +43,11 @@
 # Compilation specifics
 
 KERNEL_HEADERS := $(topdir)/libcap/include
-IPATH += -I$(topdir)/libcap/include -I$(KERNEL_HEADERS)
 
-CC := gcc
-CFLAGS := -O2
+CC ?= gcc
+CFLAGS ?= -O2
 BUILD_CC := $(CC)
-BUILD_CFLAGS := $(CFLAGS) $(IPATH)
+BUILD_CFLAGS ?= $(CFLAGS) $(IPATH)
 AR := ar
 RANLIB := ranlib
 DEBUG = -g #-DDEBUG
@@ -57,12 +56,15 @@
         -Wstrict-prototypes -Wmissing-prototypes \
         -Wnested-externs -Winline -Wshadow
 LD=$(CC) -Wl,-x -shared
-LDFLAGS := #-g
+LDFLAGS ?= #-g
 
-SYSTEM_HEADERS = /usr/include
+LIBCAP_CPPFLAGS = -I$(topdir)/libcap/include -I$(KERNEL_HEADERS)
+CPPFLAGS += $(LIBCAP_CPPFLAGS)
+BUILD_CPPFLAGS += $(LIBCAP_CPPFLAGS)
 INCS=$(topdir)/libcap/include/sys/capability.h
 LDFLAGS += -L$(topdir)/libcap
-CFLAGS += -Dlinux $(WARNINGS) $(DEBUG) $(IPATH)
+CFLAGS += $(WARNINGS)
+CPPFLAGS += -Dlinux
 PAM_CAP := $(shell if [ -f /usr/include/security/pam_modules.h ]; then echo yes ; else echo no ; fi)
 INDENT := $(shell if [ -n "$(which indent 2>/dev/null)" ]; then echo "| indent -kr" ; fi)
 DYNAMIC := $(shell if [ ! -d "$(topdir)/.git" ]; then echo yes; fi)
diff -Naur libcap-2.16-orig/libcap/Makefile libcap-2.16/libcap/Makefile
--- libcap-2.16-orig/libcap/Makefile	2008-12-04 07:03:12.000000000 +0000
+++ libcap-2.16/libcap/Makefile	2008-12-24 23:00:34.000000000 +0000
@@ -16,9 +16,10 @@
 # no support).
 ifeq ($(LIBATTR),yes)
 FILES += cap_file
-LDFLAGS += -lattr
+LDLIBS += -lattr
 endif
 
+CFLAGS+= -fPIC
 INCLS=libcap.h cap_names.h $(INCS)
 OBJS=$(addsuffix .o, $(FILES))
 MAJLIBNAME=$(LIBNAME).$(VERSION)
@@ -33,7 +34,7 @@
 endif
 
 _makenames: _makenames.c cap_names.list.h
-	$(BUILD_CC) $(BUILD_CFLAGS) $< -o $@
+	$(BUILD_CC) $(BUILD_CFLAGS) $(BUILD_CPPFLAGS) $< -o $@
 
 cap_names.h: _makenames
 	./_makenames > cap_names.h
@@ -50,15 +51,15 @@
 	$(RANLIB) $@
 
 $(MINLIBNAME): $(OBJS)
-	$(LD) $(CFLAGS) $(LDFLAGS) -Wl,-soname,$(MAJLIBNAME) -o $@ $^
+	$(LD) $(CFLAGS) $(LDFLAGS) -Wl,-soname,$(MAJLIBNAME) -o $@ $^ $(LDLIBS)
 	ln -sf $(MINLIBNAME) $(MAJLIBNAME)
 	ln -sf $(MAJLIBNAME) $(LIBNAME)
 
 %.o: %.c $(INCLS)
-	$(CC) $(CFLAGS) -c $< -o $@
+	$(CC) $(CFLAGS) $(CPPFLAGS) -c $< -o $@
 
 cap_text.o: cap_text.c $(USE_GPERF_OUTPUT) $(INCLS)
-	$(CC) $(CFLAGS) $(INCLUDE_GPERF_OUTPUT) -c $< -o $@
+	$(CC) $(CFLAGS) $(CPPFLAGS) $(INCLUDE_GPERF_OUTPUT) -c $< -o $@
 
 install: all
 	mkdir -p -m 0755 $(INCDIR)/sys
diff -Naur libcap-2.16-orig/pam_cap/Makefile libcap-2.16/pam_cap/Makefile
--- libcap-2.16-orig/pam_cap/Makefile	2008-10-17 04:31:01.000000000 +0000
+++ libcap-2.16/pam_cap/Makefile	2008-12-24 22:25:50.000000000 +0000
@@ -3,7 +3,7 @@
 topdir=$(shell pwd)/..
 include ../Make.Rules
 
-LDLIBS += -lcap
+LDLIBS += -lcap -lpam
 CFLAGS += -fPIC
 
 all: pam_cap.so
@@ -14,13 +14,13 @@
 	install -m 0755 pam_cap.so $(LIBDIR)/security
 
 pam_cap.so: pam_cap.o
-	$(LD) $(LDFLAGS) -o pam_cap.so $< $(LDLIBS)
+	$(LD) $(CFLAGS) $(LDFLAGS) -o pam_cap.so $< $(LDLIBS)
 
 pam_cap.o: pam_cap.c
-	$(CC) $(CFLAGS) -c $< -o $@
+	$(CC) $(CFLAGS) $(CPPFLAGS) -c $< -o $@
 
 testcompile: test.c pam_cap.o
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $+ -lpam -ldl $(LDLIBS)
+	$(CC) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) -o $@ $+ -lpam -ldl $(LDLIBS)
 
 clean:
 	rm -f *.o *.so testcompile *~
diff -Naur libcap-2.16-orig/progs/Makefile libcap-2.16/progs/Makefile
--- libcap-2.16-orig/progs/Makefile	2008-11-24 05:33:33.000000000 +0000
+++ libcap-2.16/progs/Makefile	2008-12-24 23:55:55.000000000 +0000
@@ -22,7 +22,7 @@
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $< $(LDLIBS)
 
 %.o: %.c $(INCS)
-	$(CC) $(CFLAGS) -c $< -o $@
+	$(CC) $(CFLAGS) $(CPPFLAGS) -c $< -o $@
 
 install: all
 	mkdir -p -m 0755 $(SBINDIR)
