--- DeviceKit-power-014/configure.ac	2010-01-08 10:49:06.000000000 +0000
+++ DeviceKit-power-014.NEW/configure.ac	2010-05-25 17:59:39.000658307 +0000
@@ -1,7 +1,7 @@
 dnl Process this file with autoconf to produce a configure script.
 
-AC_PREREQ(2.59c)
-AC_INIT(DeviceKit-power, 013, http://lists.freedesktop.org/mailman/listinfo/devkit-devel)
+AC_PREREQ(2.65)
+AC_INIT([DeviceKit-power],[013],[http://lists.freedesktop.org/mailman/listinfo/devkit-devel])
 AM_INIT_AUTOMAKE(DeviceKit-power, 014)
 AM_CONFIG_HEADER(config.h)
 AM_MAINTAINER_MODE
@@ -27,7 +27,7 @@
 AC_SUBST(LT_REVISION)
 AC_SUBST(LT_AGE)
 
-AC_ISC_POSIX
+AC_SEARCH_LIBS([strerror],[cposix])
 AC_PROG_CC
 AM_PROG_CC_STDC
 AC_HEADER_STDC
@@ -83,8 +83,7 @@
 	default_strict=no
 fi
 
-AC_ARG_ENABLE(strict, AC_HELP_STRING([--enable-strict],
-		[Enable strict compilation options]), enable_strict=$enableval,
+AC_ARG_ENABLE(strict, AS_HELP_STRING([--enable-strict],[Enable strict compilation options]), enable_strict=$enableval,
 		enable_strict=$default_strict)
 if test x$enable_strict != xno; then
 	if test "$GCC" = "yes"; then
@@ -202,7 +201,7 @@
 AM_GLIB_GNU_GETTEXT
 AC_DEFINE_UNQUOTED([GETTEXT_PACKAGE],["$GETTEXT_PACKAGE"],[gettext domain])
 
-AC_OUTPUT([
+AC_CONFIG_FILES([
 Makefile
 devkit-power-gobject.pc
 src/Makefile
@@ -220,6 +219,7 @@
 devkit-power-gobject/dkp-version.h
 po/Makefile.in
 ])
+AC_OUTPUT
 
 dnl ==========================================================================
 echo "
--- DeviceKit-power-014/doc/Makefile.am	2010-01-08 10:49:06.000000000 +0000
+++ DeviceKit-power-014.NEW/doc/Makefile.am	2010-05-25 17:57:43.782854590 +0000
@@ -72,7 +72,8 @@
 	$(NULL)
 
 if ENABLE_GTK_DOC
-include $(top_srcdir)/gtk-doc.make
+include /usr/share/gtk-doc/data/gtk-doc.make
+EXTRA_DIST =
 else
 EXTRA_DIST =
 endif
